src/Raven.Server/Documents/Indexes/Persistence/IndexOperationBase.cs

```diff
@@ -96,7 +96,7 @@ protected Query GetLuceneQuery(DocumentsOperationContext context, QueryMetadata
                         }
 
                         using (closeServerTransaction)
-                            documentQuery = QueryBuilder.BuildQuery(serverContext, context, metadata, whereExpression, _index, parameters, analyzer, factories);
+                            documentQuery = LuceneQueryBuilder.BuildQuery(serverContext, context, metadata, whereExpression, _index, parameters, analyzer, factories);
```

Comment:
Please check why we still have `GetLuceneQuery()` here, in the base class.

Reply:
Because moving them requires creating multiple classes (because IndexOperationBase is inhered by IndexReadOperationBase etc).